pauthabielf64/pauthabielf64.rst

```diff
@@ -397,100 +399,130 @@ General Principles
-  | signing schema           | to the linker with relocations and symbols         | used by the compiler in dynamic relocations.            | the signing schema specified by the linker.        | to re-sign entry from GOT.                         |
-  |                          |                                                    |                                                         |                                                    |                                                    |
-  |                          | The assembler would need new operators to describe |                                                         |                                                    |                                                    |
-  |                          | this. For example :got@auth("signing schema"):     |                                                         |                                                    |                                                    |
-  +--------------------------+----------------------------------------------------+---------------------------------------------------------+----------------------------------------------------+----------------------------------------------------+
-
-Questions/Observations
+will be writeable for the lifetime of the program. There is scope for
+some or all of the pointers in the GOT to be signed.
+
+If a pointer in the GOT is signed then the dynamic linker must sign the
+pointer at load time, and the code that loads the pointer from the GOT
+must authenticate it using the same signing schema.
+
+The ABI recommends that if RELRO is available the GOT is not
+signed. The standard GOT-generating from AAELF64_ can be used for all
+pointers in the GOT.
+
+PAUTHELF64 describes a default signing schema for GOT entries and
+AUTH variant GOT-generating relocations that can be used to create
+signed pointers in the GOT.
+
+Default signing schema
 ^^^^^^^^^^^^^^^^^^^^^^
-* If RELRO is available the ABI recommends not signing the GOT and
-  disabling lazy binding. This simplifies the work in the dynamic
-  loader. Feedback is requested on whether a signed GOT is needed.
-* If the GOT is signed there seems little advantage to the explicit
-  schema, with the disadvantage of added complexity in the tools to
-  support the new relocations. There is a possibility of a hybrid
-  model where data pointers follow an implicit signing schema and
-  function pointers in the GOT use the explicit schema. Feedback is
-  requested on whether it is necessary to sign the GOT and whether an
-  implicit signing schema would be sufficient.
+
+Signed GOT entries use the ``IA`` key for symbols of type ``STT_FUNC``
+and the ``DA`` key for symbols of , with the address of the GOT entry
```

Comment:
Incomplete sentence

Reply:
Thanks for pointing that out, I'll fix in the next update. I'm proposing IA for STT_FUNC and DA for everything else (STT_OBJECT, STT_NONE , STT_SECTION etc.)